File: Lainnya/movie-catalogue-jetpack-2-main/movie-catalogue-jetpack-2-main/movie-catalogue-jetpack-2-main/app/src/main/java/com/mramirid/moviecatalogue/ui/movies/MoviesFragment.java
package com.mramirid.moviecatalogue.ui.movies;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ProgressBar;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.lifecycle.ViewModelProviders;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import com.mramirid.moviecatalogue.R;
import com.mramirid.moviecatalogue.adapter.ItemsAdapter;
import com.mramirid.moviecatalogue.data.entity.ItemEntity;
import com.mramirid.moviecatalogue.utils.SpacesItemDecoration;
import com.mramirid.moviecatalogue.viewmodel.ViewModelFactory;

import java.util.ArrayList;

public class MoviesFragment extends Fragment {

	private RecyclerView rvMovies;
	private ProgressBar progressBar;

	public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
		return inflater.inflate(R.layout.fragment_movies, container, false);
	}

	@Override
	public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
		super.onViewCreated(view, savedInstanceState);
		rvMovies = view.findViewById(R.id.rv_movies);
		progressBar = view.findViewById(R.id.progress_bar);
	}

	@Override
	public void onActivityCreated(@Nullable Bundle savedInstanceState) {
		super.onActivityCreated(savedInstanceState);
		if (getActivity() != null) {
			MoviesViewModel viewModel = obtainViewModel(this);
			ItemsAdapter moviesAdapter = new ItemsAdapter(getActivity());
			progressBar.setVisibility(View.VISIBLE);

				viewModel.getMovies().observe(this, (ArrayList<ItemEntity> itemEntities) -> {
				progressBar.setVisibility(View.GONE);
				moviesAdapter.setItems(itemEntities);
				moviesAdapter.notifyDataSetChanged();
			});

			int spacingInPixels = getResources().getDimensionPixelSize(R.dimen.rv_item_margin);
			rvMovies.addItemDecoration(new SpacesItemDecoration(spacingInPixels));
			rvMovies.setLayoutManager(new LinearLayoutManager(getContext()));
			rvMovies.setHasFixedSize(true);
			rvMovies.setAdapter(moviesAdapter);
		}
	}

	private MoviesViewModel obtainViewModel(Fragment fragment) {
		ViewModelFactory factory = ViewModelFactory.getInstance();
		return ViewModelProviders.of(fragment, factory).get(MoviesViewModel.class);
	}
}

File: Lainnya/cielo/JetpackSubmission3v2/app/src/main/java/com/gracielo/jetpacksubmission3v2/MoviesFragment.kt
package com.gracielo.jetpacksubmission3v2

import android.content.Intent
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Toast
import androidx.fragment.app.Fragment
import androidx.lifecycle.LiveData
import androidx.lifecycle.ViewModelProvider
import androidx.paging.PagedList
import androidx.recyclerview.widget.LinearLayoutManager
import com.gracielo.jetpacksubmission3v2.Adapter.MovieAdapter
import com.gracielo.jetpacksubmission3v2.Data.Local.Entity.FilmEntity
import com.gracielo.jetpacksubmission3v2.Data.Local.Entity.MovieEntity
import com.gracielo.jetpacksubmission3v2.Movies.MovieAdapters
import com.gracielo.jetpacksubmission3v2.Movies.MovieCallback
import com.gracielo.jetpacksubmission3v2.Movies.MoviesViewModel
import com.gracielo.jetpacksubmission3v2.databinding.FragmentMoviesBinding
import com.gracielo.jetpacksubmission3v2.vo.Resource
import com.gracielo.jetpacksubmission3v2.vo.Status
import java.util.*


class MoviesFragment : Fragment(),MovieCallback {
    private lateinit var binding: FragmentMoviesBinding


    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        binding = FragmentMoviesBinding.inflate(inflater, container, false)
        return binding.root
    }


    override fun onActivityCreated(savedInstanceState: Bundle?) {
        super.onActivityCreated(savedInstanceState)

        if (activity != null) {
            val viewModel = obtainViewModel(this)
            binding.rvMovies.apply {
                layoutManager = LinearLayoutManager(context)
                adapter = MovieAdapters(this@MoviesFragment)
            }
            viewModel.getMovies().observe(viewLifecycleOwner, { listMovie ->
                if (listMovie != null) {
                    when (listMovie.status) {
                        Status.SUCCESS -> {
                            binding.rvMovies.adapter?.let { adapters ->
                                when (adapters) {
                                    is MovieAdapters -> {
                                        adapters.submitList(listMovie.data)
                                        adapters.notifyDataSetChanged()
                                    }
                                }
                            }
                        }
                        Status.ERROR -> {
                            Toast.makeText(context, "Check your internet connection", Toast.LENGTH_SHORT).show()
                        }
                    }
                }
            })

        }
    }

    private fun obtainViewModel(fragment: Fragment): MoviesViewModel {
        val factory: ViewModelFactory? = ViewModelFactory.getInstance(requireActivity())
        return ViewModelProvider(fragment, factory!!).get(MoviesViewModel::class.java)
    }

    override fun onItemClicked(data: MovieEntity) {
        startActivity(
            Intent(
                context,
                DetailActivity::class.java
            )
                .putExtra(DetailActivity.TYPE_FILM, "Movies")
                .putExtra(DetailActivity.ID_FILM, data.id)
        )
    }


}